homeassistant/components/zha/core/device.py

```diff
@@ -121,10 +121,15 @@ def __init__(
         self._checkins_missed_count = 0
         self.unsubs: list[Callable[[], None]] = []
         self.quirk_applied = isinstance(self._zigpy_device, zigpy.quirks.CustomDevice)
-        self.quirk_class = (
-            f"{self._zigpy_device.__class__.__module__}."
-            f"{self._zigpy_device.__class__.__name__}"
-        )
+
+        # Allow overriding of quirk_class for unit tests
```

Comment:
This doesn't belong in `zha/core/device.py`, us `unittest.mock within the unit test or directly overwrite the attribute on the object within the test.